contributors/design-proposals/pod-priority-api.md

```diff
@@ -0,0 +1,237 @@
+optional description. The description is an arbitrary string and is provided
+only as a guideline for users.
+
+If a pod does not specify any "PriorityClassName", the system resolves it to 
+zero in the first version, but we will add support to annotate one of the
+`PriorityClass` objects that tells the system that it is the default one. When
+there is a default priority class, a pod with no "PriorityClassName" will get
+the value of this default class (instead of zero).
+
+```
+type PriorityClass struct {
+  metav1.TypeMeta
+  // +optional
+  metav1.ObjectMeta
+  
+  // The value of this priority class. This is the actual priority that pods
+  // recieve when they have the above name in their pod spec.
+  Value        int32
+  
+  // A description of this priority class, such as the recommended use cases.
+  // +optional
+  Description  string
+}
+```
+
+And the annotation for the default class will be something like:
+
+```yaml
+...
+annotations:
+  priorityclass.alpha.kubernetes.io/is-default-class: "true"
+...
+```
+
+### Ordering of priorities 
+
+As mentioned earlier, a PriorityClassName is resolved by the admission controller to
+its integral value and Kubernetes components use the integral value. The higher
+the value, the higher the priority.
+
+### `system` PriorityClassName
+`system` is a special priority name which is reserved and cannot be changed. It
+is used for critical system pods that must never be preempted. We set default
+policies that deny creation of pods with `system` priority. Cluster admins can
+authorize users or service accounts to create pods with this priority. When
+non-authorized users set PriorityClassName to `system` in their pos spec, their
+pod creation request will be rejected. For pods created by controllers, the
+service account must be authorized by cluster admins.
+
+### Modifying priority classes 
+
+Priority classes can be added or removed, but they cannot be updated. While
+Kubernetes can work fine if priority classes are changed at run-time, the change
+can be confusing to users as pods with a priority class which were created
+before the change will have a different priority value than those created after
+the change. Deletion of priority classes is allowed only when there is no pod
+with that priority class in the system.
+
+Newly added priority classes cannot have a value higher than "system" and no one
+can change the value of "system" priority class. The reason for this restriction
+is that kubernetes critical system pods will have "system" priority and no pod 
+should be able to preempt them.
+
+#### Drawbacks of changing priority names 
+
+While Kubernetes effectively allows changing priority names (by deleting and
+adding them with a different value), it should be done only when
+absolutely needed. Changing priority names has the following disadvantages:
+
+
+*   May remove config portability: pod specs written for one cluster are no
```

Comment:
I spoke with them. Since storage is a resource, they do not reject a pod whose spec specifies a non-existing storage class. Such a pod is admitted, but remains pending. So, the config portability issue is not completely addressed with their solution either.
I think we should make it very clear when user pods are rejected because the priority class does not exist and provide instructions in the error on how to find existing priority classes.